### PR TITLE
Email sent property

### PR DESCRIPTION
**Pull request by mluo24**
### Summary 

This pull request implements part 1 of the frontend email task for "newly matched", which essentially adds a new field in the database for a timestamp of when the email for "newly matched" has been sent (either the time or a null field for it hasn't been sent). The frontend takes this value in a request and displays a purple dot accordingly for whether the email has been sent or not. If the purple dot is there, that means the email has been sent and the timestamp has a value, and hovering over it will reveal the type of email sent and the date it was sent out.

### Test Plan 

This was tested by adding data manually and testing out the two states: a null timestamp field or a filled in timestamp field (done manually). The match button was also pressed again and users were moved between groups to ensure that those functionalities would not break it.

A null timestamp:
image

A non-null timestamp with the dot being hovered over, showing tooltip: 
image

### Notes 

This PR does NOT add a feature to actually be able to change the field (i.e. this does not actually implement the email sending), so it must be changed manually in firebase. There will likely be merge conflicts with https://github.com/cornell-dti/zing-lsc/pull/40

### Breaking Changes 

- Database schema change: added a new field called shareMatchEmailTimestamp to the group schema.

### CODE REVIEW

File: frontend/src/modules/EditZing/Components/EditZing.tsx [hunk 1]
```diff
@@ -47,7 +47,15 @@ export const EditZing = () => {
       .get(`${API_ROOT}${COURSE_API}/students/${courseId}`)
       .then((res: AxiosResponse<CourseStudentDataResponse>) => {
         setUnmatchedStudents(res.data.data.unmatched)
-        setStudentGroups(res.data.data.groups)
+        let groups = res.data.data.groups.map((group: Group) => {
```
Comment (Coreball) [inline frontend/src/modules/EditZing/Components/EditZing.tsx]: Is there any particular reason for using `let` here?

File: frontend/src/modules/EditZing/Components/EditZing.tsx [hunk 2]
```diff
@@ -165,9 +173,15 @@ export const EditZing = () => {
...
         let newGroups = studentGroups.concat(response.data.data.groups)
-        console.log(response.data.data.unmatched)
+        newGroups = newGroups.map((group: Group) => {
+          return {
+            ...group,
+            shareMatchEmailTimestamp: group.shareMatchEmailTimestamp
+              ? new Date(group.shareMatchEmailTimestamp)
+              : null,
+          }
+        })
```
Comment (Coreball) [inline frontend/src/modules/EditZing/Components/EditZing.tsx]: I think newGroups might be a misnomer because in the line above it is actually `studentGroups.concat(response.data.data.groups)`. Coincidentally it still works because `new Date()` can take in an existing Date object, but we could make it clearer by only converting the new response data into Date objects.

**Review by Coreball — APPROVED** [review]
Looks very good! Getting the tooltips in as part of this PR is a nice addition too. Everything worked as I expected when testing, and using null to represent when an email hasn't been sent yet is a nice way to express it concisely. Thanks for getting this done so quickly.

File: frontend/src/modules/EditZing/Components/EditZing.tsx (hunk 1 above)
Comment (mluo24) [reply frontend/src/modules/EditZing/Components/EditZing.tsx]: Nope, good catch

File: frontend/src/modules/EditZing/Components/EditZing.tsx (hunk 2 above)
Comment (mluo24) [reply frontend/src/modules/EditZing/Components/EditZing.tsx]: Yeah I realize that now, I changed it so it only maps with the returned groups response.
